broker/config_manager.py

```diff
@@ -0,0 +1,288 @@
+"""Module providing the functionality powering the `broker config` command."""
+
+import importlib
+from importlib.metadata import version
+import json
+import os
+from pathlib import Path
+import pkgutil
+import sys
+from tempfile import NamedTemporaryFile
+
+import click
+from logzero import logger
+from packaging.version import Version
+from ruamel.yaml import YAML, YAMLError
+
+from broker import exceptions
+
+yaml = YAML()
+yaml.default_flow_style = False
+yaml.sort_keys = False
+
+C_SEP = "."  # chunk separator
+GH_CFG = "https://raw.githubusercontent.com/SatelliteQE/broker/master/broker_settings.yaml.example"
+
+
+def file_name_to_ver(file_name):
+    """Convert a version-encoded filename `v0_6_0` to a `Version` object."""
+    return Version(file_name[1:].replace("_", "."))
+
+
+class ConfigManager:
+    """Class to interact with Broker's configuration file.
+
+    One important concept of these commands is the concept of a "chunk".
+
+    A chunk is a part of the configuration file that can be accessed or updated.
+    Chunks are specified by their keys in the configuration file.
+    Nested chunks are separated by periods.
+
+    e.g. broker config view AnsibleTower.instances.my_instance
+    """
+
+    version = version("broker")
+
+    def __init__(self, settings_path=None):
+        self._interactive_mode = None
+        self._settings_path = settings_path
+        if settings_path:
+            if settings_path.exists():
+                self._cfg = yaml.load(self._settings_path)
+            else:
+                click.secho(
+                    f"Broker settings file not found at {settings_path.absolute()}.", fg="red"
+                )
+                self.init_config_file()
+
+    @property
+    def interactive_mode(self):
+        """Determine if Broker is running in interactive mode."""
+        if self._interactive_mode is None:
+            self._interactive_mode = False
+            # GitHub action context
+            if "GITHUB_WORKFLOW" not in os.environ:
+                # determine if we're being ran from a CLI
+                if sys.stdin.isatty():
+                    self._interactive_mode = True
```

Comment:
Doesn't `sys.stdin.isatty()` return False if it is being run from GH Actions?
```suggestion
            # determine if we're being run from a CLI
            if sys.stdin.isatty():
                self._interactive_mode = True
```

Reply:
unsure, but could test it by adding this to the workflow temporarily
```
python -c "import sys; print(f'{sys.stdin.isatty()=}')"
```

Reply:
update: it is indeed false, based on that test.
```
sys.stdin.isatty()=False
```

Reply:
and now completely removed that property in favor of a class attribute, thanks!